invenio_global_search/components.py

```diff
@@ -0,0 +1,174 @@
+        record: RDMRecord,
+        func: Callable = map_metadata_from_a_to_b,
+        serializer_cls=None,
+        metadata_cls=None,
+        schema=None,
+        identity=None,
+    ) -> None:
+        """Construct."""
+        self._record = record
+        self._func = func
+        self._serializer_cls = serializer_cls
+        self._metadata_cls = metadata_cls
+        self._schema = schema
+        self._identity = identity
+
+    def on_post_commit(self, uow) -> None:  # noqa: ARG002
+        """Post commit."""
+        self._func(
+            self._record,
+            self._serializer_cls,
+            self._metadata_cls,
+            self._schema,
+            self._identity,
+        )
+
+
+class Marc21ToDublinCoreComponent(ServiceComponent):
+    """Marc21ToDublinCoreComponent."""
+
+    def create(self, identity, data=None, record=None, draft=None, errors=None):
+        """Create."""
+        print(
+            f"components.py create data: {data}, draft: {draft}, record:{record}, errors: {errors}"
+        )
+        record_serializer = Marc21RecordJSONSerializer()
+        marc21 = Marc21Metadata(json=data["metadata"])
+        # data = record.dumps()
+        metadata = record_serializer.dump_obj(marc21)
+        print(f"components.py create metadata: {metadata}")
+
+    def update_draft(self, identity, data=None, record=None, errors=None):
+        """Update draft."""
+        print(f"components.py update_draft data: {data}")
+        record_serializer = Marc21RecordJSONSerializer()
+        data = record.dumps()
+        marc21 = Marc21Metadata(json=data["metadata"])
+        metadata = record_serializer.dump_obj(marc21)
+        print(f"components.py update_draft metadata: {metadata}")
+
+    def edit(self, identity, draft=None, record=None):
+        """Edit."""
+        # print(f"components.py edit draft: {draft}, record: {record}")
+
+    def publish(
+        self,
+        identity: Identity,
+        data: dict | None = None,  # noqa: ARG002
+        record: RDMRecord | None = None,
+        **_: dict,
+    ) -> None:
+        """Create handler."""
+        cmp_op = ComponentOp(
+            record,
+            serializer_cls=Marc21RecordJSONSerializer,
+            metadata_cls=Marc21Metadata,
+            schema="marc21",
+            identity=identity,
+        )
+        self.uow.register(cmp_op)
+
+
+class LOMToDublinCoreComponent(ServiceComponent):
+    """LOMToDublinCoreComponent."""
+
+    def publish(
+        self,
+        identity: Identity,
+        data: dict | None = None,  # noqa: ARG002
+        record: RDMRecord | None = None,
+        **_: dict,
+    ) -> None:
+        """Create handler."""
+        cmp_op = ComponentOp(
+            record,
+            serializer_cls=LOMRecordJSONSerializer,
+            metadata_cls=LOMMetadata,
+            schema="lom",
+            identity=identity,
+        )
+        self.uow.register(cmp_op)
+
+
+class RDMToDublinCoreComponent(ServiceComponent):
```

Comment:
Move to an instance specific config module, `app-rdm` or `invenio.cfg`.

Reply:
why?

Reply:
you mean the component should be moved to the `invenio.cfg` file or the usage? it is used in `invenio.cfg`. this package does nothing without the configuration done in `invenio.cfg`

Reply:
ok, nothing is wrong, the endpoints are there, but there is also done a some configuration in `invenio-theme-tugraz`.